### PR TITLE
Security 170

### LINKED ISSUE #342: Plugin does not provide the environment variables it is supposed to
After a recent upgrade (Jenkins to 2.5 and plugin to 1.31.4) the plugin no longer provide the needed environment variables which cause the build to fail:

GitHub pull request #441 of commit 3433d2da7885066777bc177f073df87313b9947c, no merge conflicts.
[EnvInject] - Loading node environment variables.
Building in workspace /var/lib/jenkins/jobs/MySensorsArduinoLibPR/workspace
[WS-CLEANUP] Deleting project workspace...
[WS-CLEANUP] Done
Cloning the remote Git repository
Cloning repository git@github.com:mysensors/Arduino.git
 > git init /var/lib/jenkins/jobs/MySensorsArduinoLibPR/workspace # timeout=10
Fetching upstream changes from git@github.com:mysensors/Arduino.git
 > git --version # timeout=10
using GIT_SSH to set credentials GitHub SSH key account for Jenkins
 > git -c core.askpass=true fetch --tags --progress git@github.com:mysensors/Arduino.git +refs/heads/*:refs/remotes/origin/*
 > git config remote.origin.url git@github.com:mysensors/Arduino.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url git@github.com:mysensors/Arduino.git # timeout=10
Fetching upstream changes from git@github.com:mysensors/Arduino.git
using GIT_SSH to set credentials GitHub SSH key account for Jenkins
 > git -c core.askpass=true fetch --tags --progress git@github.com:mysensors/Arduino.git +refs/pull/*:refs/remotes/origin/pr/*
 > git rev-parse origin/${ghprbActualCommit}^{commit} # timeout=10
 > git rev-parse ${ghprbActualCommit}^{commit} # timeout=10
ERROR: Couldn't find any revision to build. Verify the repository and branch configuration for this job.
[WARNINGS] Skipping publisher since build result is FAILURE
[htmlpublisher] Archiving HTML reports...
[htmlpublisher] Archiving at PROJECT level /var/lib/jenkins/jobs/MySensorsArduinoLibPR/workspace/Documentation/html to /var/lib/jenkins/jobs/MySensorsArduinoLibPR/htmlreports/Doxygen_HTML
[BFA] Scanning build for known causes...
[BFA] No failure causes found
[BFA] Done. 0s
Email was triggered for: Failure - Any
Sending email for trigger: Failure - Any
Sending email to: ${ghprbActualCommitAuthorEmail}
Finished: FAILURE

No ghprb variables are present in the environment variables of the job:
_	/usr/bin/daemon
BUILD_CAUSE	GHPRBCAUSE
BUILD_CAUSE_GHPRBCAUSE	true
BUILD_DISPLAY_NAME	#134
BUILD_ID	134
BUILD_NUMBER	134
BUILD_TAG	jenkins-MySensorsArduinoLibPR-134
BUILD_URL	http://ci.mysensors.org/job/MySensorsArduinoLibPR/134/
EXECUTOR_NUMBER	4
HOME	/var/lib/jenkins
HUDSON_HOME	/var/lib/jenkins
HUDSON_SERVER_COOKIE	2792193ede3460f8
HUDSON_URL	http://ci.mysensors.org/
JENKINS_HOME	/var/lib/jenkins
JENKINS_SERVER_COOKIE	2792193ede3460f8
JENKINS_URL	http://ci.mysensors.org/
JOB_BASE_NAME	MySensorsArduinoLibPR
JOB_NAME	MySensorsArduinoLibPR
JOB_URL	http://ci.mysensors.org/job/MySensorsArduinoLibPR/
KEYMAP	sv-latin1
LANG	en_US.UTF-8
LC_ALL	sv_SE.UTF-8
LOGNAME	jenkins
MAIL	/var/mail/jenkins
NODE_LABELS	master
NODE_NAME	master
PATH	/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games
PWD	/var/lib/jenkins
ROOT_BUILD_CAUSE	GHPRBCAUSE
ROOT_BUILD_CAUSE_GHPRBCAUSE	true
SHELL	/bin/bash
SHLVL	1
USER	jenkins
XDG_RUNTIME_DIR	/run/user/110
XDG_SESSION_ID	c14

### PR DESCRIPTION
**Pull request by DavidTanner**
Replaces #336 
Fixes #342 
Fixes janinko/ghprb#412

No need to update the core version, and fixes the tests to be compatible with future core versions.

### CODE REVIEW

**Comment by DavidTanner** [comment]
@ydubreuil does this satisfy the requirements for 170?

**Comment by orrc** [comment]
Since this fix, there seems to be a lot of empty "Parameters" links being added to each build?
image

**Comment by ydubreuil** [comment]
@DavidTanner thanks for moving things forward, seems ok for sec 170

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbAdditionalParameterEnvironmentContributor.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    @Override
+    @SuppressWarnings("rawtypes")
+    public void buildEnvironmentFor(@Nonnull Run run,
+                                    @Nonnull EnvVars envs,
+                                    @Nonnull TaskListener listener) throws IOException, InterruptedException {
+        GhprbCause cause = (GhprbCause) ((Run<?, ?>) run).getCause(GhprbCause.class);
+        if (cause == null) {
+            return;
+        }
+        run.addAction(getDefaultParameters(run));
```
Comment (daniel-beck) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbAdditionalParameterEnvironmentContributor.java]: This looks like the source of the behavior observed by @orrc.

**Comment by jlebon** [comment]
@DavidTanner Shouldn't ghprb still try to set parameters, though? E.g. if the job defines `ghprbPullId` as a parameter, ghprb should set it (and if it's not a parameter, at least it will still be injected as an env var). The advantage is that things like the Parameterized Trigger Plugin's option "Current Build Parameters" can still work.

**Comment by jlebon** [comment]
See also #352.

**Pull request merged**